### DISCUSSION TITLE
Questions regarding 44.1 vs 48 sample rates.

### DISCUSSION THREAD

**Discussion by kmitchel** [pschatzmann/ESP32-A2DP#487] [Q&A] [answered]
I am feeding an ADAU1401 with the required 48 khz sample rate, however my phone shows a SBC connection at 44.1.  Is there a mechanism handling sampling rate conversion? Or is it sending empty/duplicate frames?

Additionally, for some reason I could not get the default pins to work on a esp32-wroom-32.  I eventually found an example that suggested an alternative. http://www.winko-erades.nl/using-an-esp32-and-pcm5102-as-a-bluetooth-audio-receiver/  

Thanx.

**Comment by pschatzmann** [comment, marked as answer]
I suggest to have a look at my audio tools project that is mentioned in the readme.
There you can easily upsample the output, but please note that the default pins used by the audio tools are different.
Further information can be found in the Wiki in the chapters
- Resampling
- Converting the Format

I guess some of your pins must be damaged: you can use any gpio __output pins__!

**Reply by kmitchel** [reply]
TY...I'll do that.  Filtering on analog inputs of adau1401 are working as expected, but i2s is being really weird.

**Comment by pschatzmann** [comment]
In this case you could also use the analog output of the ESP32 instead of I2S
